admin/app/dfp/SessionWrapper.scala

```diff
@@ -103,16 +103,16 @@ private[dfp] class SessionWrapper(dfpSession: DfpSession) {
     services.networkService.getCurrentNetwork.getEffectiveRootAdUnitId
   }
 
-  def getReportQuery(reportId: Long): ReportQuery = {
+  def getReportQuery(reportId: Long): Option[ReportQuery] = {
     // Retrieve the saved query.
     val stmtBuilder = new StatementBuilder()
       .where("id = :id")
       .limit(1)
       .withBindVariableValue("id", reportId)
 
     val page: SavedQueryPage = services.reportService.getSavedQueriesByStatement(stmtBuilder.toStatement)
-    val savedQuery: SavedQuery = page.getResults(0)
-    savedQuery.getReportQuery
+    val savedQuery: Option[SavedQuery] = page.getResults().toList.headOption
```

Comment:
Won't this throw an NPE if no results?
Perhaps `Option(page.getResults())...`

Reply:
Good spot, thanks